types/src/write_set.rs

```diff
@@ -389,10 +358,7 @@ impl TransactionWrite for WriteOp {
     fn from_state_value(maybe_state_value: Option<StateValue>) -> Self {
         match maybe_state_value {
             None => Self::legacy_deletion(),
-            Some(state_value) => {
-                let (metadata, data) = state_value.unpack();
-                Self::Modification { data, metadata }
-            },
+            Some(state_value) => Self::Modification(state_value),
```

Comment:
Can you change `as_state_value()` few lines above? Just cloning the underlying state value should be enough